nvflare/app_common/widgets/validation_json_generator.py

```diff
@@ -64,11 +64,22 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
                         if data_client not in self._val_results:
                             self._val_results[data_client] = {}
                         self._val_results[data_client][model_owner] = dxo.data
+                    elif dxo.data_kind == DataKind.COLLECTION:
+                        # The DXO could contain multiple sub-DXOs (e.g. received from a T2 system)
+                        _dxos = dxo.data
+                        for _sub_data_client, _dxo in _dxos.items():
```

Comment:
Should use get_leaf_dxos() utility method to process the DXO collection. This works for single or collection DXO of any level deep.

See how it is used in CrossSiteModelEval controller.